### PR TITLE
Fixing bugs

### PR DESCRIPTION
**Pull request by krulis-martin**

### CODE REVIEW

File: app/V1Module/presenters/SubmitPresenter.php [hunk 1]
```diff
@@ -122,6 +123,13 @@ class SubmitPresenter extends BasePresenter {
    */
   public $assignmentSolutionViewFactory;
 
+  /**
+   * @var AssignmentSolutionSubmissionViewFactory
+   * @inject
+   */
+  public $assignmentSolutionSubmissionViewFactory;
+
+
```
Comment (Neloop) [inline app/V1Module/presenters/SubmitPresenter.php]: Two newlines

File: app/V1Module/presenters/SubmitPresenter.php [hunk 2]
```diff
@@ -322,6 +322,9 @@ private function finishSubmission(AssignmentSolution $solution, bool $isDebug =
     $submission->setResultsUrl($resultsUrl);
     $this->assignmentSubmissions->persist($submission);
 
+    // The solution needs to reload submissions ...
+    $this->solutions->refresh($solution);
```
Comment (Neloop) [inline app/V1Module/presenters/SubmitPresenter.php]: What exactly is not up to date here?

Comment (krulis-martin) [reply app/V1Module/presenters/SubmitPresenter.php]: The list of submissions.

Comment (Neloop) [reply app/V1Module/presenters/SubmitPresenter.php]: Ok, so why is the new submission present in the response?

Comment (Neloop) [reply app/V1Module/presenters/SubmitPresenter.php]: And again... as stated before... this is not doctrine way

File: tests/Presenters/SubmitPresenter.phpt [hunk 3]
```diff
@@ -376,7 +376,7 @@ class TestSubmitPresenter extends Tester\TestCase
     $result = $response->getPayload();
     Assert::equal(200, $result['code']);
     Assert::equal($solutionCount, count($solutions->findAll()));
-    Assert::equal($submissionCount, $solution->getSubmissions()->count());
+    Assert::equal($submissionCount+1, $solution->getSubmissions()->count());
```
Comment (Neloop) [inline tests/Presenters/SubmitPresenter.phpt]: In the days that will come, please use spaces

File: app/V1Module/presenters/ReferenceExerciseSolutionsPresenter.php [hunk 4]
```diff
@@ -498,6 +498,7 @@ private function finishSubmission(
 
     if (count($errors) > 0) {
       $this->referenceSubmissions->flush();
+      $this->referenceSolutions->refresh($referenceSolution); // special case, it would be tedious update the entity manually
```
Comment (Neloop) [inline app/V1Module/presenters/ReferenceExerciseSolutionsPresenter.php]: The comment does not really tell why is the refresh needed.

**Pull request merged**